### PR TITLE
CMNetworkDiscovery: A resource to manage the network discovery method

### PR DESCRIPTION
**Pull request by NEllis280**
#### Pull Request (PR) description
Manages the network discovery method. At this time, only enable/disable is supported.

#### This Pull Request (PR) fixes the following issues
None.

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/41)

### CODE REVIEW

File: source/DSCResources/DSC_CMNetworkDiscovery/DSC_CMNetworkDiscovery.psm1 [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+
+        if ($Enabled -ne $state.Enabled)
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMNetworkDiscovery/DSC_CMNetworkDiscovery.psm1:91]: if ($Enabled -ne $state.Enabled)
{
    Write-Verbose -Message ($script:localizedData.SettingEnable -f $state.Enabled, $Enabled)
    
    Set-CMDiscoveryMethod -NetworkDiscovery -SiteCode $SiteCode -Enabled $Enabled
}

File: tests/Unit/CMNetworkDiscovery.tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                    Enabled  = $true
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+                Mock -CommandName Set-CMDiscoveryMethod
+            }
+
+            Context 'When Set-TargetResource runs successfully' {
+
```
Comment (rchristman89) [inline tests/Unit/CMNetworkDiscovery.tests.ps1:109]: Remove extra line after context.

File: tests/Unit/CMNetworkDiscovery.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                    Set-TargetResource @getReturnDisabled
+                    Assert-MockCalled Import-ConfigMgrPowerShellModule -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-CMDiscoveryMethod -Exactly -Times 1 -Scope It
+                }
+            }
+
+            Context 'When running Set-TargetResource should throw' {
+
```
Comment (rchristman89) [inline tests/Unit/CMNetworkDiscovery.tests.ps1:132]: Remove extra line after context.

File: tests/Unit/CMNetworkDiscovery.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+                It 'Should call expected commands and throw when Set-CMDiscoveryMethod throws' {
+                    Mock -CommandName Get-TargetResource -MockWith { $getReturnEnabled }
+                    MOck -CommandName Set-CMDiscoveryMethod -MockWith { throw }
```
Comment (rchristman89) [inline tests/Unit/CMNetworkDiscovery.tests.ps1:135]: Mock casing.

Comment (NEllis280) [reply tests/Unit/CMNetworkDiscovery.tests.ps1]: Fixed casing

File: tests/Unit/CMNetworkDiscovery.tests.ps1 (hunk 3 above)
Comment (NEllis280) [reply tests/Unit/CMNetworkDiscovery.tests.ps1]: removed

File: tests/Unit/CMNetworkDiscovery.tests.ps1 (hunk 2 above)
Comment (NEllis280) [reply tests/Unit/CMNetworkDiscovery.tests.ps1]: removed

File: source/DSCResources/DSC_CMNetworkDiscovery/DSC_CMNetworkDiscovery.psm1 (hunk 1 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMNetworkDiscovery/DSC_CMNetworkDiscovery.psm1:91]: Tested and changed

File: tests/Unit/CMNetworkDiscovery.tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                    Assert-MockCalled Set-Location -Exactly -Times 2 -Scope It
+                    Assert-MockCalled Get-TargetResource -Exactly -Times 1 -Scope It
+                    Assert-MockCalled Set-CMDiscoveryMethod -Exactly -Times 1 -Scope It
+                }
+            }
+        }
+
+        Describe 'ConfigMgrCBDsc - DSC_CMNetworkDiscovery\Test-TargetResource' -Tag 'Test' {
+            BeforeAll {
+
```
Comment (rchristman89) [inline tests/Unit/CMNetworkDiscovery.tests.ps1:145]: Extra line

Comment (NEllis280) [reply tests/Unit/CMNetworkDiscovery.tests.ps1]: removed.

**Pull request merged**